src/router/middlewares/setTenantId.ts

```diff
@@ -48,6 +48,92 @@ const getTenantIdFromAudClaim = (audClaim: any, baseUrl: string): string | undef
     return uniqTenantIds[0];
 };
 
+/**
+ * Evaluates if an all tenants scope matches with the configured one.
+ * @param userIdentity The decoded access token data.
+ * @param fhirConfig The config, which includes multi-tenant setup.
+ * @returns true access granted, false denied
+ */
+function grantAccessForAllTenants(userIdentity: any, fhirConfig: FhirConfig): boolean {
+    if (fhirConfig.multiTenancyConfig?.grantAccessAllTenantsScope) {
+        const scopes: string[] = userIdentity.scope ?? [];
+        if (scopes.includes(fhirConfig.multiTenancyConfig?.grantAccessAllTenantsScope)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+const getTenantIdFromCustomClaimValue = (tenantIdClaimValue: string, prefix?: string): string | undefined => {
+    if (!prefix) {
+        return tenantIdClaimValue;
+    }
+
+    if (tenantIdClaimValue.startsWith(prefix)) {
+        return tenantIdClaimValue.substring(prefix.length);
+    }
+    return undefined;
+};
+
+/**
+ * Evaluates if tenant id url param value is included in custom claim, or not.
+ * @param tenantIdFromCustomClaim Possible tenant id values from custom claim
+ * @param prefix The optional prefix for tenantIdClaimValue
+ * @param tenantIdFromPath The tenant id url parameter to test
+ * @returns true access granted, false denied
+ */
+function grantAccessForCustomClaim(
```

Comment:
naming, like above. in this case it returns a string not a bool, so like `getCustomClaim` or `getTenantIdFromCustomClaim`